host/esUtils/client_v6.go

```diff
@@ -18,56 +18,76 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
 
-package host
+package esUtils
 
 import (
+	"fmt"
 	"io/ioutil"
 	"time"
 
 	"github.com/olivere/elastic"
 	"github.com/stretchr/testify/suite"
 )
 
-// CreateESClient create ElasticSearch client for test
-func CreateESClient(s suite.Suite, url string) *elastic.Client {
-	var err error
+type (
+	v6Client struct {
+		client *elastic.Client
+	}
+)
+
+func newV6Client(url string) (*v6Client, error) {
 	esClient, err := elastic.NewClient(
 		elastic.SetURL(url),
 		elastic.SetRetrier(elastic.NewBackoffRetrier(elastic.NewExponentialBackoff(128*time.Millisecond, 513*time.Millisecond))),
 	)
-	s.Require().NoError(err)
-	return esClient
+	return &v6Client{
+		client: esClient,
+	}, err
 }
 
-// PutIndexTemplate put index template for test
-func PutIndexTemplate(s suite.Suite, esClient *elastic.Client, templateConfigFile, templateName string) {
+func (es *v6Client) PutIndexTemplate(s suite.Suite, templateConfigFile, templateName string) {
 	// This function is used exclusively in tests. Excluding it from security checks.
 	// #nosec
 	template, err := ioutil.ReadFile(templateConfigFile)
 	s.Require().NoError(err)
-	putTemplate, err := esClient.IndexPutTemplate(templateName).BodyString(string(template)).Do(createContext())
+	putTemplate, err := es.client.IndexPutTemplate(templateName).BodyString(string(template)).Do(createContext())
 	s.Require().NoError(err)
 	s.Require().True(putTemplate.Acknowledged)
 }
 
-// CreateIndex create test index
-func CreateIndex(s suite.Suite, esClient *elastic.Client, indexName string) {
-	exists, err := esClient.IndexExists(indexName).Do(createContext())
+func (es *v6Client) CreateIndex(s suite.Suite, indexName string) {
+	exists, err := es.client.IndexExists(indexName).Do(createContext())
 	s.Require().NoError(err)
 	if exists {
-		deleteTestIndex, err := esClient.DeleteIndex(indexName).Do(createContext())
+		deleteTestIndex, err := es.client.DeleteIndex(indexName).Do(createContext())
 		s.Require().Nil(err)
 		s.Require().True(deleteTestIndex.Acknowledged)
 	}
 
-	createTestIndex, err := esClient.CreateIndex(indexName).Do(createContext())
+	createTestIndex, err := es.client.CreateIndex(indexName).Do(createContext())
 	s.Require().NoError(err)
 	s.Require().True(createTestIndex.Acknowledged)
 }
 
-// DeleteIndex delete test index
-func DeleteIndex(s suite.Suite, esClient *elastic.Client, indexName string) {
-	deleteTestIndex, err := esClient.DeleteIndex(indexName).Do(createContext())
+func (es *v6Client) DeleteIndex(s suite.Suite, indexName string) {
+	deleteTestIndex, err := es.client.DeleteIndex(indexName).Do(createContext())
 	s.Nil(err)
 	s.True(deleteTestIndex.Acknowledged)
+
+	es.client.IndexPutSettings(indexName)
```

Comment:
Just curious, why we need to put index settings here?

Reply:
I am not 100% sure. from the code it looks like we want to change the `maxResultWindowSize ` of ElasticSearch and verify it in the integrtion tests.

Reply:
ahh looks like it shouldn't be there

Reply:
Removed. I think it's from a typo when I adding the `PutMaxResultWindow `